nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -0,0 +1,339 @@
+      configureNginx = lib.mkEnableOption ''
+        Configure nginx as reverse proxy for mastodon
+        Alternatively you can configure a reverse-proxy of your choice to serve these paths:
+
+        / -> $(nix-instantiate --eval '&lt;nixpkgs&gt;' -A mastodon.outPath)/public
+        / -> 127.0.0.1:{{ webPort }} (if there was no file in the directory above)
+        /system/ -> /var/lib/mastodon/public-system/
+        /api/v1/streaming/ -> 127.0.0.1:{{ streamingPort }}
+
+        Make sure that websockets are forwarded properly. You might want to set up caching
+        of some requests, take a look at mastodon's provided nginx configuration at
+        https://github.com/tootsuite/mastodon/blob/master/dist/nginx.conf
+      '';
+      user = lib.mkOption {
+        description = ''
+          User under which mastodon runs
+          If it is set to "mastodon", a user will be created.
+        '';
+        type = lib.types.str;
+        default = "mastodon";
+      };
+      group = lib.mkOption {
+        description = ''
+          Group under which mastodon runs
+          If it is set to "mastodon", a group will be created.
+        '';
+        type = lib.types.str;
+        default = "mastodon";
+      };
+
+      streamingPort = lib.mkOption {
+        description = "TCP port used by the mastodon-streaming service";
+        type = lib.types.port;
+        default = 55000;
+      };
+      webPort = lib.mkOption {
+        description = "TCP port used by the mastodon-web service";
+        type = lib.types.port;
+        default = 55001;
+      };
+      sidekiqPort = lib.mkOption {
+        description = "TCP port used by the mastodon-sidekiq service";
+        type = lib.types.port;
+        default = 55002;
+      };
+
+      dbUser = lib.mkOption {
+        description = "Postgres login username";
+        type = lib.types.str;
+        default = "mastodon";
+      };
+      vapidPublicKey = lib.mkOption {
+        description = ''
+          The public key used for Web Push Voluntary Application Server Identification
+
+          A keypair can be generated by running
+          cd $(nix-instantiate --eval '&lt;nixpkgs&gt;' -A mastodon.outPath); bin/rake webpush:generate_keys
+        '';
+        type = lib.types.str;
+      };
+      smtpLogin = lib.mkOption {
+        description = "SMTP login name";
+        type = lib.types.str;
+      };
+
+      secretKeyBaseFile = lib.mkOption {
+        description = ''
+          Path to file containing the secret key base
+
+          Can be generated by running
+          cd $(nix-instantiate --eval '&lt;nixpkgs&gt;' -A mastodon.outPath); bin/rake secret
+        '';
+        type = lib.types.str;
+      };
+      otpSecretFile = lib.mkOption {
+        description = ''
+          Path to file containing the OTP secret
+
+          Can be generated by running
+          cd $(nix-instantiate --eval '&lt;nixpkgs&gt;' -A mastodon.outPath); bin/rake secret
```

Comment:
Is this really the same as for `secretKeyBaseFile`?

Reply:
Yes. The secrets rake task generates an OTP secret and a base secret.